### ISSUE TITLE
organize/document source

### ISSUE THREAD

**Issue by howardchung** [yasp-dota/yasp#430]
As suggested by @bwarren2:

Organize the source files, and use an automatic documentation generator:
https://github.com/jashkenas/docco

**Comment by nickhh** [comment]
I could do this, I guess. I feel like all it would result in is a lot of broken code, and in the long run I doubt many people would care. I feel like it's not worth it; the project is already so big...

**Comment by howardc93** [comment]
well it shouldn't break any code if it's just adding comments.  Docco I think will generate nice looking documentation based on that

**Comment by nickhh** [comment]
the call is to "organize the source files" which implies moving them, not
just adding comments, which is highly prone to creating errors, especially
since literally nothing in this repo is unit tested. as for the doc
generation, i'm sure we could just use java docs or some similar, which
would require very little work to compile after the comments themselves
have been written

**Comment by howardc93** [comment]
okay, we could start by organizing.  What sections are in worst need of organization/moving?

CI should tell us if we accidentally break anything.

**Comment by nickhh** [comment]
i'm not the one who requested this issue so i have no idea. usually having
one person with a vision organize things yields the best results

CI is barely an incomplete test suit let alone satisfactory unit testing

**Comment by howardc93** [comment]
Hm.  I've taken another look and it seems like the main problem is that most of the code is concentrated at the top-level.  This mixes stuff like the individual worker processes, kue processing tasks, and random globs of functions I bundle into modules for reuse.

The point of CI is to make sure nothing breaks after we move code around.  We don't necessarily need to unit test each function.

**Comment by howardc93** [comment]
As a start, we could move all the "function" modules to a /lib folder and leave only index.js + microservice init files in the top level directory.

This would require some pretty extensive refactoring though.

**Comment by howardchung** [comment]
Something not strictly related to organizing--currently we have each "helper function" module re-import it's own instance of db/redis.  This means we make a lot more connections to db/redis that we strictly need.  We should probably have those passed in as params into the function (or as part of an `options` object, so we can reuse the db/redis connection already initialized in the calling module.

**Comment by howardchung** [comment]
Is it good JS practice to enforce one function per module?  Currently a few of the "helper" modules export a bundle of related helper functions, such as `utility.js`, `queries.js`, `redis.js`, etc.

**Comment by howardchung** [comment]
I've decided I'm not going to move all the nonservice function to subdirectories.  It's really not worth the effort.  We can add comments to the functions, but it's probably much more useful just to provide an overview of what each file does, which I've done somewhat in the Developer's Guide in the README.  Automatic documentation generation isn't that useful since this isn't a library with developers needing to reference our API.

If we decided to offer parsing-as-a-service or build officially supported player/match API endpoints, we might want to document those.
